### PR TITLE
Update API for matmul

### LINKED ISSUE #201: Redefine API for matmul
## Description
- [ ] Define matmul in EncryptedTensor
- [ ] Remove n_jobs from matmul in vectors: actually not necessary at all
- [ ] bind matmul in CKKSTensor

### PR DESCRIPTION
**Pull request by youben11**
## Description
Fixes #201 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tenseal/binding.cpp [hunk 1]
```diff
@@ -283,30 +283,26 @@ PYBIND11_MODULE(_tenseal_cpp, m) {
...
-               size_t n_jobs) { return obj->matmul_plain(matrix, n_jobs); },
-            py::arg("matrix"), py::arg("n_jobs") = 0)
-        .def(
-            "mm_",
-            [](shared_ptr<CKKSVector> obj, const vector<vector<double>> &matrix,
-               size_t n_jobs) {
-                return obj->matmul_plain_inplace(matrix, n_jobs);
-            },
-            py::arg("matrix"), py::arg("n_jobs") = 0)
+        .def("matmul",
```
Comment (bcebere) [inline tenseal/binding.cpp:286]: Why not 
```
.def("matmul", &CKKSVector::matmul_plain)
```
?

File: tenseal/binding.cpp [hunk 2]
```diff
@@ -407,18 +403,16 @@ PYBIND11_MODULE(_tenseal_cpp, m) {
...
-               size_t n_jobs) { return obj->matmul_plain(matrix, n_jobs); },
-            py::arg("matrix"), py::arg("n_jobs") = 0)
-        .def(
-            "__imatmul__",
-            [](shared_ptr<CKKSVector> obj, const vector<vector<double>> &matrix,
-               size_t n_jobs) {
-                return obj->matmul_plain_inplace(matrix, n_jobs);
-            },
-            py::arg("matrix"), py::arg("n_jobs") = 0)
+        .def("__matmul__",
```
Comment (bcebere) [inline tenseal/binding.cpp:406]: same here
do we need the dedicated lambda function?

File: tenseal/binding.cpp [hunk 3]
```diff
@@ -506,6 +500,14 @@ PYBIND11_MODULE(_tenseal_cpp, m) {
...
         .def("dot", &CKKSTensor::dot_plain)
         .def("dot_", &CKKSTensor::dot_plain_inplace)
+        .def("matmul", &CKKSTensor::matmul)
+        .def("matmul_", &CKKSTensor::matmul_inplace)
+        .def("matmul", &CKKSTensor::matmul_plain)
+        .def("matmul_", &CKKSTensor::matmul_plain_inplace)
+        .def("mm", &CKKSTensor::matmul)
+        .def("mm_", &CKKSTensor::matmul_inplace)
+        .def("mm", &CKKSTensor::matmul_plain)
+        .def("mm_", &CKKSTensor::matmul_plain_inplace)
```
Comment (bcebere) [inline tenseal/binding.cpp:510]: Shouldn't we expose these in ckkstensor.py as well?

**Review by bcebere — APPROVED** [review]
Great work! I've left some questions.

File: tenseal/tensors/ckkstensor.py [hunk 4]
```diff
@@ -92,6 +92,22 @@ def dot_(self, other) -> "CKKSTensor":
         self.data.dot_(other)
         return self
 
+    def mm(self, other) -> "CKKSTensor":
```
Comment (bcebere) [inline tenseal/tensors/ckkstensor.py:95]: One more thing: we should have tests for this too. 
if not in this PR, maybe open an issue, to remember to test it.

Comment (youben11) [reply tenseal/tensors/ckkstensor.py:95]: done

**Pull request merged**